components/resource-creation/ResourceDisplay.tsx

```diff
@@ -292,9 +221,51 @@ function ResourceDisplay() {
     }
   };
 
-  const handleSortedResources = (sorted: fhir4.BundleEntry[]) => {
-    setSortedResources(sorted);
-  };
+  // Card Filtering
+  useEffect(() => {
+    if (selectedPatient && selectedDataRequirement && currentTestCases[selectedPatient].resources.length > 0) {
+      // Search
+      const filtered = currentTestCases[selectedPatient].resources.filter(entry => {
+        if (!entry.resource) return false;
+        const resourceType = entry.resource.resourceType?.toLowerCase() || '';
+        const label = getFhirResourceSummary(entry.resource).toLowerCase();
+        const { date, dateType } = dateForResource(entry.resource);
+        return (
+          resourceType.includes(cardFilters.searchString) ||
+          label.includes(cardFilters.searchString) ||
+          date.toLowerCase().includes(cardFilters.searchString) ||
+          dateType.toLowerCase().includes(cardFilters.searchString)
+        );
+      });
+
+      // Sort
+      if (cardFilters.sortType) {
+        filtered.sort((a, b) => {
+          if (!a.resource || !b.resource) return 0;
+          if (cardFilters.sortType === 'date') {
+            const dateA = dateForResource(a.resource).date.toLowerCase();
+            const dateB = dateForResource(b.resource).date.toLowerCase();
+            return cardFilters.sortOrder === 'asc' ? dateA.localeCompare(dateB) : dateB.localeCompare(dateA);
+          }
+          if (cardFilters.sortType === 'resourceType') {
+            const typeA = a.resource.resourceType.toLowerCase();
+            const typeB = b.resource.resourceType.toLowerCase();
+            return cardFilters.sortOrder === 'asc' ? typeA.localeCompare(typeB) : typeB.localeCompare(typeA);
+          }
+          return 0;
+        });
+      }
+
+      setFilteredPatientResources(filtered);
+    }
+  }, [
+    cardFilters.searchString,
+    cardFilters.sortType,
+    cardFilters.sortOrder,
```

Comment:
Recommend simplifying these 3 to just `cardFilters` rather than calling out each field individually.